library/system.lisp

```diff
@@ -61,12 +63,15 @@ While the result will always contain microseconds, some implementations may retu
                (cl:* 1000000 (cl:- end start))
                cl:internal-time-units-per-second)))))
 
-  (declare sleep (Integer -> Unit))
+  (declare sleep ((math:Rational :a) => :a -> Unit))
   (define (sleep n)
     "Sleep for `n` seconds."
```

Comment:
Add documentation about what n can be in a follow on paragraph, per your PR description. That was very useful design information!